### PR TITLE
Add ParseURL method

### PR DESCRIPTION
**Pull request by szubtsovskiy**
Added a method to parse database URL to use in environments like Heroku (described in #445) including some sanity checks.

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -3,9 +3,13 @@ package pg
 import (
 	"crypto/tls"
 	"net"
+	"net/url"
 	"time"
 
+	"errors"
+	"fmt"
 	"gopkg.in/pg.v5/internal/pool"
+	"strings"
```
Comment (vmihailenco) [inline options.go]: Please regroup imports - https://github.com/golang/go/wiki/CodeReviewComments#imports

File: options.go [hunk 2]
```diff
@@ -103,6 +107,71 @@ func (opt *Options) init() {
 	}
 }
 
+func ParseURL(sURL string) (*Options, error) {
+
```
Comment (vmihailenco) [inline options.go]: Remove empty line?

File: options.go [hunk 3]
```diff
@@ -103,6 +107,71 @@ func (opt *Options) init() {
...
+	parsedUrl, err := url.Parse(sURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// scheme
+	if parsedUrl.Scheme != "postgres" {
+		return nil, errors.New("Invalid scheme: " + parsedUrl.Scheme)
```
Comment (vmihailenco) [inline options.go]: Can we add `pg: ` prefix to these errors? E.g. `fmt.Errorf("pg: invalid scheme: %s", parsedUrl.Scheme)`

File: options_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
+// +build go1.7
```
Comment (vmihailenco) [inline options_test.go]: Why limit this to go1.7?

**Comment by vmihailenco** [comment]
Overall looks good - left some minor comments.

File: options_test.go (hunk 4 above)
Comment (szubtsovskiy) [inline options_test.go]: Copy & paste :-)

File: options.go (hunk 1 above)
Comment (szubtsovskiy) [inline options.go]: Done

File: options.go (hunk 2 above)
Comment (szubtsovskiy) [inline options.go]: Done

File: options.go (hunk 3 above)
Comment (szubtsovskiy) [inline options.go]: Done

File: options_test.go (hunk 4 above)
Comment (szubtsovskiy) [inline options_test.go]: Fixed

**Comment by szubtsovskiy** [comment]
I've amended and force pushed the commit

**Comment by vmihailenco** [comment]
Awesome, thanks.

**Pull request merged**